src/preload.py

```diff
@@ -51,7 +51,7 @@
     "flash-boot",  # flasher images
 ]
 
-SUPERVISOR_REPOSITORY_RE = "^resin(playground)?/[a-z0-9]+-supervisor$"
+SUPERVISOR_REPOSITORY_RE = "^(balena|resin)?(playground)?/?[a-z0-9]+-supervisor$"
```

Comment:
This will match an invalid repo named `playground`:
```
$ python
>>> from re import match
>>> SUPERVISOR_REPOSITORY_RE = "^(balena|resin)?(playground)?/?[a-z0-9]+-supervisor$"

# unexpected - matches invalid repo 'playground'
>>> not not match(SUPERVISOR_REPOSITORY_RE, 'playground/amd64-supervisor')
True
```
A small improvement would be:
```suggestion
SUPERVISOR_REPOSITORY_RE = "^((balena|resin)(playground)?/)?[a-z0-9]+-supervisor$"
```

```
$ python
>>> from re import match
>>> SUPERVISOR_REPOSITORY_RE = "^((balena|resin)(playground)?/)?[a-z0-9]+-supervisor$"

>>> not not match(SUPERVISOR_REPOSITORY_RE, 'playground/amd64-supervisor')
False

>>> not not match(SUPERVISOR_REPOSITORY_RE, 'balena/amd64-supervisor')
True

>>> not not match(SUPERVISOR_REPOSITORY_RE, 'balenaplayground/amd64-supervisor')
True
```

With or without this improvement, it would still match some strange / invalid names as follows, but further improvement might not be worth (?) the complexity like keeping track of all valid architecture strings:
```
# missing slash
>>> not not match(SUPERVISOR_REPOSITORY_RE, 'balenaamd64-supervisor')
True

# invalid architecture
>>> not not match(SUPERVISOR_REPOSITORY_RE, 'playground-supervisor')
True
```

Reply:
@pdcastro there are not so many architectures so we can do:
```
SUPERVISOR_REPOSITORY_RE = "^(?:(?:balena|resin|balenaplayground)??/)??(?:armel|rpi|armv7hf|aarch64|i386|amd64|i386-nlp)-supervisor$"
```
Note that there is no `resinplayground` in dockerhub so I omitted it.